elasticgraph-schema_artifacts/lib/elastic_graph/schema_artifacts/runtime_metadata/enum.rb

```diff
@@ -17,8 +17,17 @@ module Enum
         # Runtime metadata related to an ElasticGraph enum type.
         #
         # @private
-        class Type < ::Data.define(:values_by_name)
+        class Type < ::Data.define(:values_by_name, :grouping_missing_value_placeholder)
           VALUES_BY_NAME = "values_by_name"
+          MISSING_VALUE_PLACEHOLDER = "**missing**"
+
+          def initialize(values_by_name:)
+            has_conflict = values_by_name.any? do |name, value|
+              name == MISSING_VALUE_PLACEHOLDER || value.datastore_value == MISSING_VALUE_PLACEHOLDER
```

Comment:
It's not clear to me why this checks `datastore_value` but not `datastore_abbreviation`.

...although, I'm not convinced that we have to check either--take a look at how `datastore_value`/`datastore_abbreviation` are used:

* https://github.com/search?q=repo%3Ablock%2Felasticgraph%20%22.datastore_value%22&type=code
* https://github.com/search?q=repo%3Ablock%2Felasticgraph+%22.datastore_abbreviation%22&type=code

Neither of them are used as values returned by Elasticsearch/OpenSearch in a grouping. Instead, they are values we pass *to* Elasticsearch/OpenSearch in a query.

I might be missing something, though--if there's any chance there could be a collision we should leave this (and maybe add a check on `datastore_abbreviation` as well).

Reply:
Since `datastore_value` and `datastore_abbreviation` aren't used in a way that would conflict with the missing value placeholder, I'm no longer doing any collision detection on enum.